golang/internal/helper/docker/container.go

```diff
@@ -0,0 +1,132 @@
+package docker
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/filters"
+	"github.com/docker/docker/client"
+	"github.com/rs/zerolog/log"
+
+	"github.com/dyrector-io/dyrectorio/golang/internal/dogger"
+)
+
+const DockerClientTimeoutSeconds = 30
+
+func DeleteContainerByName(ctx context.Context, dog *dogger.DeploymentLogger, nameFilter string) error {
+	matchedContainer, err := GetContainerByName(ctx, nil, nameFilter)
+	if err != nil {
+		return fmt.Errorf("builder could not get container (%s) to remove: %s", nameFilter, err.Error())
+	}
+
+	switch matchedContainer.State {
+	case "running", "paused", "restarting":
+		err = StopContainerByName(ctx, dog, nameFilter)
+		if err != nil {
+			return fmt.Errorf("builder could not stop container (%s): %s", nameFilter, err.Error())
+		}
+		fallthrough
+	case "exited", "dead", "created":
+		err = RemoveContainerByName(ctx, dog, nameFilter)
+		if err != nil {
+			return fmt.Errorf("builder could not remove container (%s): %s", nameFilter, err.Error())
+		}
+		return nil
+	case "":
+		// when there's no container we just skip it
+		return nil
+	default:
+		return fmt.Errorf("builder could not determine the state (%s) of the container (%s) for deletion: %s",
+			matchedContainer.State,
+			nameFilter,
+			err.Error())
+	}
+}
+
+func StopContainerByName(ctx context.Context, dog *dogger.DeploymentLogger, nameFilter string) error {
+	cli, err := client.NewClientWithOpts(client.FromEnv, client.WithAPIVersionNegotiation())
+	if err != nil {
+		log.Fatal().Err(err).Msg("")
+	}
+
+	timeoutValue := (time.Duration(DockerClientTimeoutSeconds) * time.Second)
+	if dog != nil {
+		dog.Write("Container request for: " + nameFilter)
+	}
+	if err := cli.ContainerStop(ctx, nameFilter, &timeoutValue); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Matches one
+func RemoveContainerByName(ctx context.Context, dog *dogger.DeploymentLogger, nameFilter string) error {
```

Comment:
Why is it public?

Reply:
This is a helper lib. Functions have to be public, there is chance we do not use all of them currently but I'd like to keep the opportunity open.

Reply:
Yeah but in this case we should only use the DeleteContainerByName function, or am i missing something?

Reply:
We only use that one because most of the time we do "I don't care i just want to see this container GONE" and this is what that function does. 

If we would want to stop a running or remove a stopped container we would have to explicitly use these. These aren't in our scope right now, but this might change in the future.